python-frontend/src/main/java/org/sonar/python/semantic/v2/converter/ClassDescriptorToPythonTypeConverter.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * SonarQube Python Plugin
+ * Copyright (C) 2011-2024 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.python.semantic.v2.converter;
+
+import org.sonar.python.index.ClassDescriptor;
+import org.sonar.python.index.Descriptor;
+import org.sonar.python.semantic.v2.ClassTypeBuilder;
+import org.sonar.python.types.v2.LazyTypeWrapper;
+import org.sonar.python.types.v2.Member;
+import org.sonar.python.types.v2.PythonType;
+
+public class ClassDescriptorToPythonTypeConverter implements DescriptorToPythonTypeConverter {
+
+  private static PythonType convert(ConversionContext ctx, ClassDescriptor from) {
+    var typeBuilder = new ClassTypeBuilder()
+      .withName(from.name())
+      .withDefinitionLocation(from.definitionLocation());
+
+    from.superClasses().stream()
+      .map(fqn -> ctx.lazyTypesContext().getOrCreateLazyType(fqn))
+      .map(LazyTypeWrapper::new)
```

Comment:
```suggestion
      .map(fqn -> ctx.lazyTypesContext().getOrCreateLazyTypeWrapper(fqn))
```

I would use the wrapper instead